File: src/main/java/ru/ssau/tourism/entities/Payment.java
package ru.ssau.tourism.entities;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Setter
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "voucher_id")
    private Voucher voucher;
    @Column
    private LocalDate payDay;
    @Column
    private Integer amount;

    public Long getId() {
        return id;
    }

    public Voucher getVoucher() {
        return voucher;
    }

    public LocalDate getPayDay() {
        return payDay;
    }

    public Integer getAmount() {
        return amount;
    }
}

File: src/main/kotlin/ru/ssau/tourism/entities/Payment.kt
package ru.ssau.tourism.entities

import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate
import javax.persistence.*

@Entity
class Payment constructor(
		@Id
		@GeneratedValue(strategy = GenerationType.AUTO)
		var id: Long? = null,
		@ManyToOne
		@JoinColumn(name = "voucher_id")
		var voucher: Voucher? = null,
		@Column
		@DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
		var payDay: LocalDate? = null,
		@Column
		var amount: Int? = null
) {
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other?.javaClass != javaClass) return false
		other as Payment
		if (id != other.id) return false
		if (payDay != other.payDay) return false
		return true
	}

	override fun hashCode(): Int {
		var result = id?.hashCode() ?: 0
		result = 31 * result + (payDay?.hashCode() ?: 0)
		return result
	}
}
